File: DeliveryExpress.Contracts/CreateDeliveryRequest/CreateDeliveryRequest.cs
using DeliveryExpress.Contracts.Common;

namespace DeliveryExpress.Contracts.CreateDeliveryRequest
{
    public class CreateDeliveryRequest
    {
        public int ClientId { get; set; }

        public Address Address { get; set; } = null!;
        public IEnumerable<DeliveryItems> Items { get; set; } = null!;
    }

    public class DeliveryItems
    {
        public int Product { get; set; }
        public int Quantity { get; set; }
    }
}

File: src/@core/delivery/usecases/create-delivery/create-delivery.request.ts
import {
  ArrayMinSize,
  IsNotEmptyObject,
  IsOptional,
  Min,
} from 'class-validator';
import { AddressRequest } from 'src/@core/address/address.request';

export class CreateDeliveryRequest {
  @IsOptional()
  description: string;
  @Min(1)
  clientId: number;
  @Min(1)
  stablishmentId: number;
  @ArrayMinSize(1)
  deliveryItems: [
    {
      productId: number;
      quantity: number;
    },
  ];
  @IsNotEmptyObject()
  address: AddressRequest;
}
